com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java

```diff
@@ -157,9 +159,23 @@ private void removeOldOrphanedFiles(Date expirationTime) {
     }
 
     private void removeActivitiHistoricData(Date expirationTime) {
+        Set<String> historicProcessIds = new HashSet<String>();
         activitiFacade.getHistoricProcessInstancesFinishedAndStartedBefore(expirationTime)
             .stream()
-            .forEach(historicProcessInstance -> activitiFacade.deleteHistoricProcessInstance(historicProcessInstance.getId()));
+            .forEach(historicProcessInstance -> addProcessAndSubProcesses(historicProcessIds, historicProcessInstance.getId()));
+
+        removeHistoricProcesses(historicProcessIds);
+    }
+
+    private void addProcessAndSubProcesses(Set<String> historicProcessIds, String historicProcessId) {
+        historicProcessIds.add(historicProcessId);
+        List<String> subProcessIds = activitiFacade.getHistoricSubProcessIds(historicProcessId);
+        historicProcessIds.addAll(subProcessIds);

```

Comment:
The method activitiFacade.getHistoricSubProcessId uses the variables which are related to the correlationId. Such variables indeed are part of the sub-processes but the correlationId is also added for the parent process. 
Could you please check that?
Ha, you have used a Set here.... smart but please check it :)

Reply:
getHistoricSubProcessId has a check to ensure the parent process is not included in the list of ids:
if (!historicVariableInstance.getProcessInstanceId().equals(superProcessId))

Reply:
yes, right :)